src/System.Linq.Parallel/tests/QueryOperators/AverageTests.cs

```diff
@@ -77,6 +79,16 @@ public static void Average_Long_Longrunning(Labeled<ParallelQuery<int>> labeled,
         }
 
         [Theory]
+        [MemberData("Ranges", (object)(new int[] { 2 }), MemberType = typeof(UnorderedSources))]
+        public static void Average_Long_Overflow(Labeled<ParallelQuery<int>> labeled, int count)
```

Comment:
We don't have a test like that, no.  I tried writing one like that, but it didn't work, because it casts directly to a double for the final calculation; it never goes back to int.

If you're thinking of `Sum`, you had me split that off to a separate issue ( #2316 ).